modules/installation-azure-confidential-vms.adoc

```diff
@@ -0,0 +1,79 @@
+
+You can enable Confidential VMs for compute nodes, control plane nodes, or all nodes when installing your cluster. Confidential VMs can operate in two modes: 
+
+* Only encrypting the virtual machine guest state storage, which contains the security state of the virtual machine
+* Encrypting the virtual machine guest state storage and the operating system storage
+
+If you encrypt the operating system storage, you can use a platform-managed encryption key or a key you manage.
+
+[IMPORTANT]
+====
+You can use confidential VMs with the following VM sizes:
+
+* DCasv5-series
+* DCadsv5-series
+* ECasv5-series
+* ECadsv5-series
+
+Confidential VMs are currently not supported on 64-bit ARM architectures.
+====
+
+.Prerequisites
+* You have created an `install-config.yaml` file.
+
+.Procedure
+
+* Use a text editor to edit the install-config.yaml file prior to deploying your cluster and add one of the following stanzas:
+.. To use confidential VMs that only encrypt the virtual machine guest state:
++
+[source,yaml]
+----
+controlPlane: <1>
+  platform:
+    azure:
+      settings:
+        securityType: ConfidentialVM <2>
+        confidentialVM:
+          uefiSettings:
+            secureBoot: Enabled <3>
+            virtualizedTrustedPlatformModule: Enabled <4>
+      osDisk:
+        securityProfile:
+          securityEncryptionType: VMGuestStateOnly <5>
+----
+<1> Specify `controlPlane.platform.azure` or `compute.platform.azure` to deploy confidential VMs on only control plane or compute nodes respectively. Specify `platform.azure.defaultMachinePlatform` to deploy confidential VMs on all nodes.
+<2> Enable confidential VMs. 
+<3> Enable secure boot. For more information, see Azure's documentation about link:https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch#secure-boot[secure boot].
+<4> Enable the virtualized Trusted Platform Module. For more information, see Azure's documentation about link:https://learn.microsoft.com/en-us/windows/security/hardware-security/tpm/trusted-platform-module-overview[virtualized Trusted Platform Modules].
+<5> Specify `VMGuestStateOnly` to encrypt the VM guest state.
++
+.. To use confidential VMs that encrypt both the VM guest state and the OS disk:
++
+[source,yaml]
+----
+controlPlane:
+  platform:
+    azure:
+      settings:
+        securityType: ConfidentialVM
+        confidentialVM:
+          uefiSettings:
+            secureBoot: Enabled
+            virtualizedTrustedPlatformModule: Enabled
+      osDisk:
+        securityProfile:
+          securityEncryptionType: DiskWithVMGuestState <1>
+          diskEncryptionSet: <2>
+            resourceGroup: <your-resource-group-name>
+            name: <your-des-name>
+            subscriptionId: <subscription-uuid>
+----
+<1> Enable OS disk and VM guest state encryption.
+<2> Specify disk encryption set parameters for user-managed encryption, or omit the `diskEncryptionSet` stanza for platform-managed encryption.
```

Comment:
when using user-managed encryption, the encryption type of `diskEncryptionSet` must be `ConfidentialVmEncryptedWithCustomerKey`, the default one is `EncryptionAtRestWithPlatformKey` .

And based on this [guide](https://learn.microsoft.com/en-us/azure/confidential-computing/quick-create-confidential-vm-azure-cli-amd?tryIt=true&source=docs#create-confidential-virtual-machine-using-a-customer-managed-key), it might also need to add option `--enabled-for-disk-encryption` when creating keyvault. And testing is blocked here ( related bug [OCPBUGS-18379](https://issues.redhat.com/browse/OCPBUGS-18379)), we could not verify the correct steps of creating diskEncryptionSet for confidentialVM.

It should be not the same as existing procedure [here](https://docs.openshift.com/container-platform/4.13/installing/installing_azure/enabling-user-managed-encryption-azure.html).

Reply:
Jinyun is right about the process of creating a `diskEncryptionSet` for Confidential VMs being different than the existing [one](https://docs.openshift.com/container-platform/4.13/installing/installing_azure/enabling-user-managed-encryption-azure.html) and at the same time, we were not able to verify the respective procedure for CVMs (not only with RHCOS, but also with a compatible CVM version of Ubuntu).  

@bscott-rh  since we have yet to define a resolution time for [OCPBUGS-18379](https://issues.redhat.com/browse/OCPBUGS-18379), i.e. we are currently not sure when we will be able to support the `DiskWithVMGuestState`  encryption type, perhaps we could skip documenting the `DiskWithVMGuestState` (both with platform-managed and user-managed encryption) encryption type altogether.  Maybe we could just write that support for this encryption type is pending and coming in the next releases of OCP? (is there any guideline about how to document features in such state or should we skip documenting it?)